### PR TITLE
#2 copied over more repo settings

### PR DESCRIPTION
**Pull request by ShravanSunder**
- more settings
- github actions

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import './test/helpers/hardhat-imports';
+
+import { HardhatUserConfig } from 'hardhat/types';
+import { mochaRootHook_disableReact18Warnings } from 'test/helpers/mochaRootHooks';
```
Comment (stevenpslade) [inline hardhat.config.ts]: We do not have `@typechain/hardhat` under `devDependencies` (I had removed hardhat packages in my last PR). If we are copying over this file maybe it's time we just bite the bullet and take all the deps from `eth-hooks`.

We probably also need `test/helpers/hardhat-imports` and `test/helpers/mochaRootHooks` copied over as well (and maybe `/test-files/__mocks__/`?).

I think this is more inline with your original suggestion anyways of just taking it all from `eth-hooks` 💥

Comment (ShravanSunder) [reply hardhat.config.ts]: perhaps we don't need hardhat as we'll mock the underlying hooks from eth-hooks for tests? what do you think @stevenpslade ?

Comment (stevenpslade) [reply hardhat.config.ts]: Yeah that's what I was thinking when I originally omitted `hardhat.config.ts` - I think we can get all that from `eth-hooks`.

Comment (stevenpslade) [reply hardhat.config.ts]: In that case I think almost everything else in this PR is fine: we just remove `hardhat.config.ts` and the couple of lines it's referenced in a few other files.

**Pull request merged**